extensions/interactions/ItemSelectionInput/directives/oppia-short-response-item-selection-input.component.ts

```diff
@@ -45,11 +45,10 @@ export class ShortResponseItemSelectionInputComponent implements OnInit {
     ) as string[];
     const choices = this.htmlEscaperService.escapedJsonToObj(
       this.choices
-    ) as { _html: string; _contentId: string }[];
+    ) as string[];
 
-    const choicesContentIds = choices.map(choice => choice._contentId);
     this.responses = answer.map(
-      contentId => choices[choicesContentIds.indexOf(contentId)]._html);
+      contentId => choices[parseInt(contentId.split('_')[2])]);
```

Comment:
Could you add some explanation of what is going on here? It may be making an assumption about the content ID that may or may not always hold.

Reply:
As I mentioned above, the choices array is not an array of objects of type `{_html: string; _contentId: string}`, but rather an array of strings (which is just the _html property). This means `_contentId` cannot be used for indexing through the choices array.

I'm thus obtaining the last character of each `contentId string present in answer array, passing it through the parseInt() function to obtain a number, which I can then use as an index for the choices array.

For example, say the answer array contains the values [`"ca_choice_0"`, `"ca_choice_1"`]. I perform the split and parseInt operations to obtain 0, and 1, which I use as indices in the choices array to obtain the required choices.

Additionally, I don't have to specifically retrieve the _html property, because, like I said, the array no longer consists of objects.

Please let me know if this answers your question!

> assumption about the content ID that may or may not always hold

I see, I wasn't aware of this. Is it possible the values stored in the answer array are not stored in order according to the last character of each string? i.e. is "ca_choices_1" not guaranteed to correspond to the second choice of the choices array? I haven't encountered such a case, but I'll look into it.

Reply:
(I'm still looking into this)

Reply:
I think you may run into issues if you e.g. delete some choice options and add more later, since the content IDs are supposed to be unique throughout the state (and they move together with the answers they are attached to). Perhaps you could try that.

Reply:
You're right, I tried deleting and adding some choices and the indexing got messed up. Clearly, there is no way to reliably index through the choices array without the contentID values.

This problem arises with interactions that have the choices property in their `interactionCustomizationArgs` (where `getAnswerHtml` maps the entire choice object, but `getShortAnswerHtml` maps only the _html property of each object).

![image](https://user-images.githubusercontent.com/78612244/158802715-9cb6167e-3893-45b0-8550-06096bbbb889.png)

I can think of two ways I could work around this
 - I could map the entire choice object in `getShortAnswerHtml` as well(similar to how it is done in `getAnswerHtml`), and then individually adjust all interactions with the `choices` property that will get affected by this change, or,
 - I could add a check for the interactionID, and if it happens to be `ItemSelectionInput`, the function will map the entire choice object, and if not, it will only map the _html property, like so:
 
![image](https://user-images.githubusercontent.com/78612244/158803766-2752003a-044c-4d76-b090-400d737f21b4.png)
 
 This will ensure other interactions with a choices property (such as multiple choice) will remain unaffected.
 
 I would like to know which of these sound like a better option to you or perhaps if you have any other alternatives in mind.
 
 @seanlip PTAL!

Reply:
I feel like I don't understand what the issue is, actually.

This is a directive that displays the user's answer, right? You already have the user's answer. Why not just display that (what's the need for "choices")?

Reply:
By user do you mean the learner? The choices selected by a learner are displayed using `getAnswerHtml()`. They are displayed just fine and do not have any issue that I'm aware of (apart from the one where selecting no answer causes console errors, and I've already made the required changes to fix it, up above).

However the piece of code in question does not correspond to `getAnswerHtml()` -- it corresponds to `getShortAnswerHtml()` which is used _not_ to display the learner's choices but to display the solution added by the creator.

The `answer` array that we have access to stores the `contentId`s of the correct choices picked by the creator. We use these `contentId`s to pick the right choices (if the `contentId` of a choice in the choices array, is present in the `answer` array, the choice is considered to be correct and the `_html` properties of all such correct choices are used in the solution modal).

To clarify, when a creator adds a solution to an interaction of type ItemSelectionInput, its not the right choices picked by them that get stored as `answer` -- it's the `contentId`s of whatever choice(s) the creator selected as the right choice(s), that get stored as `answer`.

We later (i.e. when displaying the solution) use these `contentId`s in the answer array and compare them against the `contentId`s in the choices array (and this is why need the choices array) to determine the right choices.

Please let me know if this answers your question.

Reply:
Ah sorry, I was thinking of the answer rather than the short answer. Sorry about that. (As an aside -- what does the UI for the answer display look like now when no choices are selected -- do we show a message for that rather than just leaving the whole thing blank?)

In answer to your previous question, I would go with the first approach (to handle "short answer" similarly to "answer"). We should not be making special-case exceptions to particular interactions in the general code.

Reply:
> what does the UI for the answer display look like now when no choices are selected

This text is displayed when the learner hits submit with no choices selected:
![image](https://user-images.githubusercontent.com/78612244/159130689-238a2387-c194-4c4a-97fc-c3dcc8b3da9e.png)


> I would go with the first approach (to handle "short answer" similarly to "answer")

Got it, thanks!

Reply:
That UI looks good. Thanks!